### PR TITLE
add arm64 to the build

### PR DESCRIPTION
**Pull request by tcurdt**
The builds system is rather complex and I couldn't really test this - but I hope it should work like that?
Let me know.

### CODE REVIEW

File: contrib/build-dokku.Dockerfile [hunk 1]
```diff
@@ -24,6 +24,7 @@ COPY . ${WORKDIR}
 
 ENV GOPATH=/go
 
+
```
Comment (josegonzalez) [inline contrib/build-dokku.Dockerfile:27]: Mind removing all these extra newlines you added?

File: contrib/build-dokku.Dockerfile [hunk 2]
```diff
@@ -59,4 +61,26 @@ RUN PLUGIN_MAKE_TARGET=${PLUGIN_MAKE_TARGET} \
   GOARCH=arm make version copyfiles \
   && DOKKU_ARCHITECTURE=armhf GOARCH=arm make deb-dokku
 
+
+FROM builder as arm64
+
+COPY --from=amd64 /tmp /tmp
```
Comment (josegonzalez) [inline contrib/build-dokku.Dockerfile:67]: Probably should copy from `armhf` instead, as that will include the amd64 files.

File: contrib/build-dokku.Dockerfile [hunk 3]
```diff
@@ -59,4 +61,26 @@ RUN PLUGIN_MAKE_TARGET=${PLUGIN_MAKE_TARGET} \
...
+COPY --from=amd64 /usr/local/share/man/man1/dokku.1 /usr/local/share/man/man1/dokku.1-generated
```
Comment (josegonzalez) [inline contrib/build-dokku.Dockerfile:68]: This can stay as amd64.

File: contrib/release-dokku [hunk 4]
```diff
@@ -415,6 +417,8 @@ main() {
     git checkout -- plugins/*/plugin.toml
   fi
 
+  # what about this?
+  # echo "build/dokku_${NEXT_VERSION}_arm64.deb" >build/deb-filename
```
Comment (josegonzalez) [inline contrib/release-dokku:421]: Drop these two lines.

**Review by josegonzalez — CHANGES_REQUESTED** [review]

File: contrib/build-dokku.Dockerfile (hunk 1 above)
Comment (tcurdt) [reply contrib/build-dokku.Dockerfile:27]: Sure can - I added them to make the file a little bit more readable.

File: contrib/build-dokku.Dockerfile (hunk 2 above)
Comment (tcurdt) [reply contrib/build-dokku.Dockerfile:67]: It's all the same, no?

Comment (tcurdt) [reply contrib/build-dokku.Dockerfile:67]: Ups.

File: contrib/release-dokku (hunk 4 above)
Comment (tcurdt) [reply contrib/release-dokku:421]: What's the idea behind those other two filename lines?

File: contrib/build-dokku.Dockerfile (hunk 2 above)
Comment (tcurdt) [reply contrib/build-dokku.Dockerfile:67]: `arm64` you mean.

File: contrib/release-dokku (hunk 4 above)
Comment (josegonzalez) [reply contrib/release-dokku:421]: They are used by tests to extract the correct file. The tests run in amd64.

File: contrib/build-dokku.Dockerfile (hunk 2 above)
Comment (tcurdt) [reply contrib/build-dokku.Dockerfile:67]: Hm. Shouldn't it be
```
FROM builder as armhf

COPY --from=armhf /tmp /tmp
COPY --from=amd64 /usr/local/share/man/man1/dokku.1 /usr/local/share/man/man1/dokku.1-generated
...
FROM builder as arm64

COPY --from=arm64 /tmp /tmp
COPY --from=amd64 /usr/local/share/man/man1/dokku.1 /usr/local/share/man/man1/dokku.1-generated
```

File: contrib/build-dokku.Dockerfile [hunk 5]
```diff
@@ -41,7 +41,7 @@ RUN PLUGIN_MAKE_TARGET=${PLUGIN_MAKE_TARGET} \
 
 FROM builder as armhf
 
-COPY --from=amd64 /tmp /tmp
+COPY --from=armhf /tmp /tmp
```
Comment (josegonzalez) [inline contrib/build-dokku.Dockerfile:44]: Please revert this. You're _in_ the `armhf` build stage. We're trying to grab the `/tmp` folder _from_ the `amd64` stage.

Maybe useful reading:

- https://www.docker.com/blog/advanced-dockerfiles-faster-builds-and-smaller-images-using-buildkit-and-multistage-builds/
- https://levelup.gitconnected.com/docker-multi-stage-builds-and-copy-from-other-images-3bf1a2b095e0

File: contrib/build-dokku.Dockerfile (hunk 2 above)
Comment (tcurdt) [reply contrib/build-dokku.Dockerfile]: OK, now I get the flow. This is incrementally adding and so chaining the folders. That was not obvious.

File: contrib/build-dokku.Dockerfile (hunk 5 above)
Comment (tcurdt) [reply contrib/build-dokku.Dockerfile]: Already did. Thanks.

Comment (tcurdt) [reply contrib/build-dokku.Dockerfile]: It wasn't clear why I would want to grab a `amd64` folder in an `armhf` stage.
That would make a useful comment in the dockerfile ;)

**Review by josegonzalez — APPROVED** [review]

**Comment by josegonzalez** [comment]
Since this still needs other dependencies, will be hard to test this without going through everything else. I don't have much time this week to do dokku-related work (maybe this weekend) so if you wanted to start on the other dependencies, that would be great. I can definitely give reviews too.

**Comment by josegonzalez** [comment]
In _theory_ things will work with arm64 once this is merged (and we'll even have arm/arm64 docker images!) but we'll need to test it then.

**Pull request merged**